util/uuid.c

```diff
@@ -1,34 +1,44 @@
-#include <uuid.h>
+#define _DEFAULT_SOURCE
 #include "util/uuid.h"
 
-#if HAS_LIBUUID
-bool generate_uuid(char out[static 37]) {
-	uuid_t uuid;
-	uuid_generate_random(uuid);
-	uuid_unparse(uuid, out);
-	return true;
-}
-#else
-#include <assert.h>
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdio.h>
 #include <string.h>
-#include <stdlib.h>
+#include <unistd.h>
+
+#ifndef HAS_GETENTROPY
+static int getentropy(void *buffer, size_t len) {
+	FILE *urandom;
+	int ret = -1;
 
-bool generate_uuid(char out[static 37]) {
-	uuid_t uuid;
-	uint32_t status;
-	uuid_create(&uuid, &status);
-	if (status != uuid_s_ok) {
+	if ((urandom = fopen("/dev/urandom", "r"))) {
+		if (fread(buffer, 1, len, urandom) != len) {
+			ret = 0;
+		}
+		fclose(urandom);
+	}
+
+	return ret;
+}
+#endif
+#define UUID_FORMAT "%04" PRIx16 "%04" PRIx16 \
+	"-%04" PRIx16 \
+	"-%04" PRIx16 \
+	"-%04" PRIx16 \
+	"-%04" PRIx16 "%04" PRIx16 "%04" PRIx16
+bool generate_uuid(char out[static UUID_STRLEN])
```

Comment:
@emersion should we just make this a 128-bit hex string instead of a UUID?

Reply:
I'm fine with either. Would it make the string formatting less awkward?

Reply:
Without any pretense of UUID formatting one can just use two 64-bit numbers `"%016" PRIx64 "%016" PRIx64` and keep everything on one line. Though that leads to the question of whether it should be renamed to reflect what it's actually doing.

Reply:
Yeah. If we go for an hex string, we could s/uuid/token/